Assets/UIElements/_Elements/XRUI_Feedback/Scripts/XRUI_Feedback.cs

```diff
@@ -145,35 +155,63 @@ private void CreateNearCollider()
                 throw new Exception("Unknow NearColliderType selected");
         }
         nearCollider.isTrigger = true;
-
         xrNearDetection = nearColliderObject.AddComponent<XRUI_NearDetectionLocked.XRUI_NearDetection>();
+
+        if (allowDistanceEvents)
+        {
+            if (xrInteractable != null)
+            {
+                xrInteractable.colliders.Add(nearCollider);
+                xrInteractable.onHoverEnter.AddListener((XRBaseInteractor) => { xrNearDetection.OnEnterAction(XRBaseInteractor); });
+                xrInteractable.onHoverExit.AddListener((XRBaseInteractor) => { xrNearDetection.OnExitAction(XRBaseInteractor); });
+            }
+        }
     }
 
     private void ConfigureSelectListeners()
     {
-        xrInteractable.onSelectEnter.AddListener((XRBaseInteractor interactor) =>
+        if (allowDistanceEvents)
         {
-            XRController controller = interactor.GetComponent<XRController>();
-            if (controller != null)
+            SetColliderOptions();
+            xrInteractable.onSelectEnter.AddListener((XRBaseInteractor interactor) =>
             {
-                isSelected = true;
-                nearCollider.gameObject.SetActive(false);
-                GetComponent<Collider>().enabled = false;
-                onSelectEnter?.Invoke(controller);
-            }
-        });
+                OnEnterInteraction(interactor);
+            });
 
-        xrInteractable.onSelectExit.AddListener((XRBaseInteractor interactor) =>
-        {
-            XRController controller = interactor.GetComponent<XRController>();
-            if (controller != null)
+            xrInteractable.onSelectExit.AddListener((XRBaseInteractor interactor) =>
             {
-                isSelected = false;
-                nearCollider.gameObject.SetActive(true);
-                GetComponent<Collider>().enabled = true;
-                onSelectExit?.Invoke(controller);               
-            }
-        });
+                OnExitInteraction(interactor);
+            });
+        }
+    }
+
+    private void SetColliderOptions()
+    {
+        Collider collider = gameObject.GetComponent<Collider>();
+        if (collider != null && collider.isTrigger)
+            collider.isTrigger = false;
+    }
+    private void OnEnterInteraction(XRBaseInteractor interactor)
+    {
+        XRController controller = interactor.GetComponent<XRController>();
+        if (controller != null)
+        {
+            isSelected = true;
+            onSelectEnter?.Invoke(controller);
+
+            XRUI_3DButtonBase button3d = GetComponentInParent<XRUI_3DButtonBase>();
+            if (button3d != null)
```

Comment:
This need to be done on the 3dButton script, because XRFeedback it's used in all elements.

A sugestion is on XRUI_3dButtonBase Awake add a function to listener:

feedback.onEnterInteraction.AddListener(.....)

Reply:
Done